### PR TITLE
add docs on viem

### PR DESCRIPTION
**Pull request by eshaben**
### Description

This PR adds docs on viem! 🎉 

Overview:
- adds a new page on viem
- updates compile snippets to support ts
- rearranges library-related snippets to mirror the directory structure

Note: to avoid confusion around which type of client to use, i have deviated a bit from the standard structure we use for the other ethereum libraries to explicitly show which provider to use for every snippet. as such, i have opted out of using the "view complete script" expandable code snippets. 

### Checklist

- [x] If this requires translations for the `moonbeam-docs-cn` repo, I have created a ticket for the translations in Jira
- [x] If pages have been moved around,  I have created an additional PR in `moonbeam-docs-cn` to move the pages
- [x] If images have been added, I have run the `compress-images.py` script to compress the images.
- [x] If this page requires a disclaimer, I have added one

### Corresponding PRs

Moved snippets: https://github.com/moonbeam-foundation/moonbeam-docs-cn/pull/339
⚠️ The CN PR must be merged right after this one so we can publish them at the same time ⚠️ 

### After Translation Requirements

- [x] No additional PRs are required after the translations are done

### CODE REVIEW

File: builders/build/eth-api/libraries/viem.md [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+--8<-- 'code/eth-api/libraries/viem/transaction.ts'
+```
+
+To run the script, you can run the following command in your terminal:
+
+```bash
+npx ts-node transaction.ts
+```
+
+If the transaction was succesful, in your terminal you'll see the transaction hash has been printed out.
```
Comment (themacexpert) [inline builders/build/eth-api/libraries/viem.md:273]:
```suggestion
If the transaction was successful, in your terminal you'll see the transaction hash has been printed out.
```

File: builders/build/eth-api/libraries/viem.md [hunk 2]
```diff
@@ -0,0 +1,416 @@
+---
+title: How to use viem Ethereum Library
+description: Check out this tutorial to learn how to use the viem TypeScript interface for Ethereum to send transactions and deploy Solidity smart contracts to Moonbeam.
+---
+
+# viem TypeScript Ethereum Library
+
+## Introduction {: #introduction }
+
+[viem](https://viem.sh/){target=_blank} is a modular TypeScript library that allows developers to interact with abstractions over the JSON-RPC API, making it easy to interact with Ethereum nodes. Since Moonbeam has an Ethereum-like API available that is fully compatible with Ethereum-style JSON RPC invocations, developers can leverage this compatibility to interact with Moonbeam nodes. For more information on viem, check out their [documentation site](https://viem.sh/docs/getting-started.html){target=_blank}.
```
Comment (themacexpert) [inline builders/build/eth-api/libraries/viem.md:10]: JSON-RPC has a hyphen in one spot and doesn't have one in another.

File: builders/build/eth-api/libraries/viem.md [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+  --8<-- 'text/faucet/faucet-list-item.md'
+- 
+  --8<-- 'text/common/endpoint-examples-list-item.md'
+
+!!! note
+    --8<-- 'text/common/assumes-mac-or-ubuntu-env.md'
+
+## Installing viem {: #installing-viem }
+
+To get started, you'll need to create a basic TypeScript project. First, create a directory to store all of the files you'll be creating throughout this guide, and initialize the project with the following command:
```
Comment (themacexpert) [inline builders/build/eth-api/libraries/viem.md:28]: I think you should specify what version of node you're using. I tried out the tutorial with v18.18.0

File: .snippets/code/eth-api/libraries/viem/deploy.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+// 1. Update imports
+import { createPublicClient, createWalletClient, http } from 'viem';
+import { privateKeyToAccount } from 'viem/accounts';
+import { moonbaseAlpha } from 'viem/chains';
+import contractFile from './compile';
+
+// 2. Create a wallet client for writing chain data
```
Comment (themacexpert) [inline .snippets/code/eth-api/libraries/viem/deploy.ts:7]: Maybe mention here in a comment that the private key needs to be prepended with the `0x`. If you omit the 0x, as MetaMask does when copying private keys, you'll get a nondescript error.

**Review by themacexpert — APPROVED** [review]
Looks great!

File: builders/build/eth-api/libraries/viem.md (hunk 2 above)
Comment (eshaben) [reply builders/build/eth-api/libraries/viem.md:10]: Good catch! I actually just fixed this in a separate PR: https://github.com/moonbeam-foundation/moonbeam-docs/pull/760

**Comment by albertov19** [comment]
Looks good from my perspective if @themacexpert approved it :)

**Pull request merged**